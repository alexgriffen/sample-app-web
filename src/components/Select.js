import React, { Component } from 'react';
import * as PropTypes from 'prop-types';
import Select from '@material-ui/core/Select';
import { withStyles } from '@material-ui/core/styles';
import { colors } from '../utils/Colors';
import filterSvg from '../assets/svg/filter3x.svg';

const styles = theme => (
  {
    root: {
      backgroundColor: colors.white,
      '&:hover': {
        //you want this to be the same as the backgroundColor above
        backgroundColor: colors.white,
      },
      '&::before': {
        content: `url(${ filterSvg })`,
        height: 40,
        width: 40,
        position: 'absolute',
        left: 0,
        top: 0,
      },
      borderRadius: 0,
    },
    select: {
      color: colors.gray,
      padding: 0,
      fontSize: 14,
      height: 40,
      textTransform: 'uppercase',
      width: 220,
      textAlign: 'center',
      textAlignLast: 'center',
      position: 'relative',
      [ theme.breakpoints.down('md') ]: {
        // Dirty little hack to hide the text but still make it clickable
        width: 0,
        paddingLeft: 40,
        minWidth: 0,
      },
    },
    icon: {
      [ theme.breakpoints.down('md') ]: {
        display: 'none',
      },
    },
  }
);
const { string, func, object } = PropTypes;

class SelectBox extends Component {
  static propTypes = {
    classes: object.isRequired,
    dataTest: string,
    onChange: func.isRequired,
    fallBackClasses: string,
    testID: string,
    options: PropTypes.arrayOf(PropTypes.shape({
      value: string.isRequired,
      label: string.isRequired,
    })).isRequired,
  };

  static defaultProps = {
    fallBackClasses: '',
  };

  render() {
    const { classes, dataTest, fallBackClasses, options, onChange, testID } = this.props;

    return (
      <Select
        classes={ {
          // ` ${fallBackClasses}` is there to make it backwards compatible,
          // it doesn't not have an other function
          root: `${ classes.root } ${ fallBackClasses }`,
          select: classes.select,
          icon: classes.icon,
        } }
        { ...(dataTest ? { 'data-test': dataTest } : {}) }
        { ...(testID ? { testID } : {}) }
        native
        onChange={ onChange }
      >
        { options.map(option =>
          <option key={ option.id } value={ option.value }>{ option.label }</option>
        ) }
      </Select>
    );
  }
}

export default withStyles(styles)(SelectBox);
