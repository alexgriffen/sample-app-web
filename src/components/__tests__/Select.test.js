import React from 'react';
import { shallow } from 'enzyme/build';
import Select from '../Select';

let props;

describe('Select', () => {
  beforeEach(() => {
    props = {
      options: [
        { value: 'az', label: 'Name (A to Z)' },
        { value: 'za', label: 'Name (Z to A)' },
        { value: 'lohi', label: 'Price (low to high)' },
        { value: 'hilo', label: 'Price (high to low)' },
      ],
      onChange: jest.fn(),
    };
  });

    it('should render correctly with the required options', () => {
      const component = shallow(
        <Select
          { ...props }
        />
      ).dive();

      expect(component).toMatchSnapshot();
    });

    it('should render correctly with full options', () => {
      const component = shallow(
        <Select
          { ...props }
          dataTest="data-test-string"
          fallBackClasses="fallback-classes"
          testID="select"
        />
      ).dive();

      expect(component).toMatchSnapshot();
    });
});
