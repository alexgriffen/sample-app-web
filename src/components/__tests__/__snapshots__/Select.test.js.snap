// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly with full options 1`] = `
<WithStyles(WithFormControlContext(Select))
  classes={
    Object {
      "root": "SelectBox-root-1 fallback-classes",
      "select": "SelectBox-select-2",
    }
  }
  data-test="data-test-string"
  native={true}
  onChange={[MockFunction]}
  testID="select"
>
  <option
    value="az"
  >
    Name (A to Z)
  </option>
  <option
    value="za"
  >
    Name (Z to A)
  </option>
  <option
    value="lohi"
  >
    Price (low to high)
  </option>
  <option
    value="hilo"
  >
    Price (high to low)
  </option>
</WithStyles(WithFormControlContext(Select))>
`;

exports[`Select should render correctly with the required options 1`] = `
<WithStyles(WithFormControlContext(Select))
  classes={
    Object {
      "root": "SelectBox-root-1 ",
      "select": "SelectBox-select-2",
    }
  }
  native={true}
  onChange={[MockFunction]}
>
  <option
    value="az"
  >
    Name (A to Z)
  </option>
  <option
    value="za"
  >
    Name (Z to A)
  </option>
  <option
    value="lohi"
  >
    Price (low to high)
  </option>
  <option
    value="hilo"
  >
    Price (high to low)
  </option>
</WithStyles(WithFormControlContext(Select))>
`;
